src/app/services/course/course.service.ts

```diff
@@ -29,4 +29,6 @@ export class CourseService {
 
   delete = (course:Course) => this.http.delete<Course>(this.path+'/api/courses/'+course.id)
 
+  deleteAll = (filter: String = ""): Observable<Course[]> => this.http.delete<Course[]>(this.path + '?query=' + filter)
```

Comment:
deleteAll(courses: string[]): Observable<Course[]> {
    let params = new HttpParams();
    courses.forEach(course => {
      params = params.append('idCourses', course);
    });

    return this.http.delete<Course[]>(`${this.path}/api/courses`, { params });
  }